reccmp/isledecomp/parser/linter.py

```diff
@@ -13,7 +14,7 @@ class DecompLinter:
     def __init__(self) -> None:
         self.alerts: list[ParserAlert] = []
         self._parser = DecompParser()
-        self._filename: str = ""
+        self._filename: PurePath = PurePath("")
```

Comment:
Is this intentionally an OS-dependent `PurePath`? Usually, you either have a `Path` (OS-dependent, real path) or one of `PurePosixPath` and `PureWindowsPath` (path that may not correspond to anything on the present system)